value.go

```diff
@@ -297,14 +301,41 @@ func makeUnboundSelfBinding() selfBinding {
 	}
 }
 
+// Hack - we need to distinguish not-checked-yet and no error situations
+// so we have a special value for no error and nil means that we don't know yet.
+var errNoErrorInObjectInvariants = errors.New("No error - assertions passed")
+
 type valueObjectBase struct {
 	valueBase
+	assertionError error
 }
 
 func (*valueObjectBase) typename() string {
 	return "object"
 }
 
+func (obj *valueObjectBase) assertionsChecked() bool {
```

Comment:
I'm a little confused here - can't an assertion be checked without producing an error?

Reply:
Ah heh, just read your "errNoError". Makes sense...

Reply:
I wish I had proper algebraic types available. Coming up with special values like that feels dirty... 

I've added another comment to minimize confusion.